test_util/launch_configs/onprem-from-info.yaml

```diff
@@ -0,0 +1,15 @@
+---
+# no deployment_name provided because it was set in the cloud_cluster
+deployment_type: onprem
+installer_url: https://downloads.dcos.io/dcos/testing/master/dcos_generate_config.sh
+provider: cloud_cluster
+provider_info_json_path: /foo/bar/important/path
+dcos_config:
+    num_private_agents: 2
+    num_public_agents: 1
+    num_masters: 3
+    cluster_name: My Awesome DC/OS
+    resolvers:
+        - 8.8.4.4
+        - 8.8.8.8
+    security: strict
```

Comment:
So, this gets a little tricky. You say:

> Instead of adding new options like `num_masters`

This is the opposite of the situation. In the *real* config.yaml, num_masters is a real term. Rather, its the only the *installer* config that requires a `master_list` which is then converted into `num_masters` in a "real" config.yaml

On the other hand, the `num_agents` and `num_public_agents` come from `agent_list` and `public_agent_list` which are *installer only* names. The "real" config.yaml doesnt care about those counts and they are just used to deploy

Finally, you have the other odd connectors:
- ip-detect. Passed to the installer via magic file path (genconf/ip-detect). Installer injects this magic file path (because it is closer in line to the legacy '--genconf' instructions). Thus, using 'detect_ip_filename' is something we can directly pass to config.yaml with impunity
- ssh_key. Passed via another magic path (genconf/ssh_key). This is an installer-only term and is already covered else where in dcos-launch config doc
- rexray: Passed via yet another magic path that is injected into config.yaml. Thus, we can just use native calc.py's 'rexray_config' which reads in a file path and coverts to contents
- detect-ip-public: The only hook we have for this is to pass the direct contents of the file. The installer does this by, guess what, another magic path. I would suggest we add to gen a detect_ip_public_filename as to maintain the same model as the regular detect-ip

So, what I am getting at here is that this would be a config file format that is less magic than the installer (which again, you need to thoroughly read through the docs to find implicit settings, rather than explicitly declaring in a config file). Only for a the agent lists does this breakdown because agent lists are not something the config cares about.

Thus, I think the question should be, at which level do we declare the agent list counts without confusing the user